src/sql/base/browser/ui/table/treeGrid.ts

```diff
@@ -0,0 +1,167 @@
+			const keyboardEvent = (<any>e).originalEvent;
+			if (keyboardEvent instanceof KeyboardEvent) {
+				let event = new StandardKeyboardEvent(keyboardEvent);
+				if (event.keyCode === KeyCode.Enter) {
+					// toggle the collapsed state of the row
+					this.expandRow(data.row);
+					return false;
+				} else if (event.keyCode === KeyCode.LeftArrow) {
+					// Left arrow on first cell of the expanded row collapses it
+					if (data.cell === 0) {
+						this.expandRow(data.row, false); // Collapsing state
+						return false;
+					}
+				} else if (event.keyCode === KeyCode.RightArrow) {
+					// Right arrow on last cell of the collapsed row expands it.
+					if (data.cell === (this._grid.getColumns().length - 1)) {
+						this.expandRow(data.row, true);
+						return false;
+					}
+				}
+			}
+			return true;
+		});
+
+		this._grid.onRendered.subscribe((e, data) => {
+			const visibleRows = this._grid.getViewport();
+			for (let i = visibleRows.top; i < visibleRows.bottom; i++) {
+				const rowData = this._data.getItem(i);
+				// Getting the row div that corresponds to the data row
+				const rowElement = this._tableContainer.querySelector(`div [role="row"][aria-rowindex="${(i + 1)}"]`);
+				// If the row element is found in the dom, we are setting the required aria attributes for it.
+				if (rowElement) {
+					if (rowData.expanded !== undefined) {
+						rowElement.ariaExpanded = rowData.expanded;
+					}
+					if (rowData.level !== undefined) {
+						rowElement.ariaLevel = rowData.level;
+					}
+					if (rowData.setSize !== undefined) {
+						rowElement.ariaSetSize = rowData.setSize;
+					}
+					if (rowData.posInSet !== undefined) {
+						rowElement.ariaPosInSet = rowData.posInSet;
+					}
+				}
+			}
+			return false;
+		});
+	}
+
+	override setData(data: Array<T>): void;
+	override setData(data: TableDataView<T>): void;
+	override setData(data: AsyncDataProvider<T>): void;
+	override setData(data: Array<T> | TableDataView<T> | AsyncDataProvider<T>): void {
+		if (data instanceof TableDataView || data instanceof AsyncDataProvider) {
+			this._data = data;
+		} else {
+			this._data = new TableDataView<T>(data, undefined, undefined, defaultTreeGridFilter);
+		}
+		this.addTreeGridDataAttributes(this._data);
+		this._grid.setData(this._data, true);
+		this._data.filter(this._grid.getColumns());
+	}
+
+	/**
+	 *  Sets the expanded state to the specified value, or if undefined toggles the current state of the cell
+	 */
+	private expandRow(row: number, expanded?: boolean): void {
+		const rowData = this._data.getItem(row);
+		if (rowData['isParent']) {
+			if (expanded === undefined) {
+				(<any>rowData).expanded = !rowData.expanded;
+			} else {
+				(<any>rowData).expanded = expanded;
+			}
+			this._data.filter(this._grid.getColumns());
+			this.rerenderGrid();
+			this.focus();
+		}
+	}
+
+	/**
+	 * We need to augment the grid data to include additional properties that are necessary for rendering a tree grid.
```

Comment:
Typically comments don't talk in first person, you just need to describe what it's doing 

`Adds additional properties to data rows necessary for displaying as part of the tree grid structure`